lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -137,6 +160,53 @@ void SPIRVRegularizeLLVM::lowerMemset(MemSetInst *MSI) {
   return;
 }
 
+void SPIRVRegularizeLLVM::buildFunnelShiftLeftFunc(Function *FSHLFunc) {
+  if (!FSHLFunc->empty())
+    return;
+
+  auto *RotateBB = BasicBlock::Create(M->getContext(), "rotate", FSHLFunc);
+  IRBuilder<> Builder(RotateBB);
+  Type *Ty = FSHLFunc->getReturnType();
+  // Build the actual funnel shift rotate logic.
+  // In the comments, "int" is used interchangeably with "vector of int
+  // elements".
+  FixedVectorType *VectorTy = dyn_cast<FixedVectorType>(Ty);
+  Type *IntTy = VectorTy ? VectorTy->getElementType() : Ty;
```

Comment:
```suggestion
  Type *ElementTy = VectorTy ? VectorTy->getElementType() : Ty;
```

Reply:
Technically, this variable gets used for a "non-element" type in the simple integer scenario....